lib/sisimai/bite/email/google.rb

```diff
@@ -0,0 +1,320 @@
+        # We recommend contacting the other email provider for further information about the
+        # cause of this error. The error that the other server returned was:
+        # 550 550 5.7.1 SPF unauthorized mail is prohibited. (state 15).
+        # 554 554 Error: no valid recipients (state 15).
+        '15' => { 'command' => 'DATA', 'reason' => 'filtered' },
+
+        # http://www.google.com/support/forum/p/Google%20Apps/thread?tid=0aac163bc9c65d8e&hl=en
+        # Technical details of permanent failure:
+        # Google tried to deliver your message, but it was rejected by the recipient domain.
+        # We recommend contacting the other email provider for further information about the
+        # cause of this error. The error that the other server returned was:
+        # 550 550 <****@***.**> No such user here (state 17).
+        # 550 550 #5.1.0 Address rejected ***@***.*** (state 17).
+        '17' => { 'command' => 'DATA', 'reason' => 'filtered' },
+
+        # Technical details of permanent failure:
+        # Google tried to deliver your message, but it was rejected by the recipient domain.
+        # We recommend contacting the other email provider for further information about the
+        # cause of this error. The error that the other server returned was:
+        # 550 550 Unknown user *****@***.**.*** (state 18).
+        '18' => { 'command' => 'DATA', 'reason' => 'filtered' },
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'Google Gmail: https://mail.google.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-Failed-Recipients']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from Google Gmail
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        # Google Mail
+        # From: Mail Delivery Subsystem <mailer-daemon@googlemail.com>
+        # Received: from vw-in-f109.1e100.net [74.125.113.109] by ...
+        #
+        # * Check the body part
+        #   This is an automatically generated Delivery Status Notification
+        #   Delivery to the following recipient failed permanently:
+        #
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [89.01/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->